pkg/cmd/pipeline/logs.go

```diff
@@ -0,0 +1,133 @@
+// Copyright © 2019 The Tekton Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pipeline
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"github.com/tektoncd/cli/pkg/cli"
+	"github.com/tektoncd/cli/pkg/cmd/pipelinerun"
+	"github.com/tektoncd/cli/pkg/helper/pods"
+	"github.com/tektoncd/cli/pkg/helper/pods/stream"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// LogOptions provides options on what logs to fetch. An empty LogOptions
+// implies fetching all logs including init steps
+type LogOptions struct {
+	pipelineName    string
+	pipelinerunName string
+	tasks           []string
+	allSteps        bool
+	follow          bool
+	stream          *cli.Stream
+	params          cli.Params
+	streamer        stream.NewStreamerFunc
+}
+
+func logCommand(p cli.Params) *cobra.Command {
+	opts := &LogOptions{params: p}
+	eg := `
+  # show the logs of the latest pipelinerun of given pipeline
+	tkn pipeline logs foo -n bar
+
+   `
+	c := &cobra.Command{
+		Use: "logs",
+		DisableFlagsInUseLine: true,
+		Short:   "Show the logs of latest pipelinerun of given pipeline ",
+		Example: eg,
+		Args:    cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts.pipelineName = args[0]
+			opts.stream = &cli.Stream{
+				Out: cmd.OutOrStdout(),
+				Err: cmd.OutOrStderr(),
+			}
+			opts.streamer = pods.NewStream
+
+			return opts.run()
```

Comment:
we can just query the last `pipelinerun` then pass it to `pipelinerun.LogOption`  then invoke Run()?

Reply:
Make sense! Done